bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/discovery/SolarwattDevicesDiscoveryService.java

```diff
@@ -0,0 +1,200 @@
+        if (localScanningJob != null && !localScanningJob.isCancelled()) {
+            localScanningJob.cancel(true);
+            this.scanningJob = null;
+        }
+    }
+
+    @Override
+    protected synchronized void startScan() {
+        this.removeOlderResults(this.getTimestampOfLastScan());
+        final EnergyManagerHandler localEnergyManagerHandler = this.energyManagerHandler;
+
+        if (localEnergyManagerHandler == null
+                || localEnergyManagerHandler.getThing().getStatus() != ThingStatus.ONLINE) {
+            this.logger.warn("Energymanager handler not available: {}", localEnergyManagerHandler);
+            return;
+        }
+        this.scanForDeviceThings();
+    }
+
+    /**
+     * Scans for device things.
+     *
+     * Walks through the list of devices and adds discovery results for the supported devices.
+     */
+    private void scanForDeviceThings() {
+        @Nullable
+        EnergyManagerHandler localEnergyManagerHandler = this.energyManagerHandler;
+        if (localEnergyManagerHandler != null) {
+            @Nullable
+            final Map<String, Device> devices = localEnergyManagerHandler.getDevices();
+
+            final ThingUID bridgeUID = localEnergyManagerHandler.getThing().getUID();
+
+            if (devices == null) {
+                this.logger.warn("No device data for solarwatt devices in discovery for energy manager {}.", bridgeUID);
+            } else {
+                devices.forEach((key, entry) -> {
+                    if (entry instanceof EnergyManager) {
+                        // energy manager is our bridge
+                    } else if (entry instanceof BatteryConverter) {
+                        this.discover(bridgeUID, entry, THING_TYPE_BATTERYCONVERTER);
+                    } else if (entry instanceof Inverter) {
+                        this.discover(bridgeUID, entry, THING_TYPE_INVERTER);
+                    } else if (entry instanceof PowerMeter) {
+                        this.discover(bridgeUID, entry, THING_TYPE_POWERMETER);
+                    } else if (entry instanceof EVStation) {
+                        this.discover(bridgeUID, entry, THING_TYPE_EVSTATION);
+                    } else if (entry instanceof Location) {
+                        this.discover(bridgeUID, entry, THING_TYPE_LOCATION);
+                    } else if (entry instanceof PVPlant) {
+                        this.discover(bridgeUID, entry, THING_TYPE_PVPLANT);
+                    } else if (entry instanceof GridFlow) {
+                        this.discover(bridgeUID, entry, THING_TYPE_GRIDFLOW);
+                    } else if (entry instanceof SmartEnergyManagement || entry instanceof SimpleSwitcher) {
+                        // deprecated class
+                        this.logger.trace("Ignoring deprecated device {}", entry.getClass().getName());
+                    } else {
+                        this.logger.info("Ignoring device {}", entry.getClass().getName());
```

Comment:
debug?
https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Reduced loglevels.

Only some error logging remains. This occurs only when everything is beyond hope and only a `restart bundle`seems to fix things.